add Ceil and Floor filters and their tests

cdwiegand opened the pull request:
Added ceil and floor liquid filters, could have used them on a recent project. Issue is https://github.com/dotliquid/dotliquid/issues/330

daviburg commented on src/DotLiquid.Tests/StandardFilterTests.cs:
```diff
@@ -764,6 +764,32 @@ public void TestRound()
+        {
+            using (CultureHelper.SetCulture("en-GB"))
+            {
+                Helper.AssertTemplateResult("2", "{{ 1.234678 | ceil }}");
```
Please cover the 4 spec examples at

https://shopify.github.io/liquid/filters/ceil/

Importantly do test the different input type i.e. number vs string containing a number.

daviburg commented on src/DotLiquid.Tests/StandardFilterTests.cs:
```diff
@@ -764,6 +764,32 @@ public void TestRound()
+        {
+            using (CultureHelper.SetCulture("en-GB"))
+            {
+                Helper.AssertTemplateResult("1", "{{ 1.234678 | floor }}");
```
Same here, cover all spec normative examples:
https://shopify.github.io/liquid/filters/floor/

daviburg commented on src/DotLiquid/StandardFilters.cs:
```diff
@@ -650,6 +650,42 @@ public static object Round(object input, object places = null)
+        {
+            try
+            {
+                var i = Convert.ToDecimal(input);
```
please inline single-use variable here and below

daviburg commented on src/DotLiquid/StandardFilters.cs:
```diff
@@ -650,6 +650,42 @@ public static object Round(object input, object places = null)
+                var i = Convert.ToDecimal(input);
+                return Math.Ceiling(i);
+            }
+            catch (Exception)
```
do not swallow all exceptions. target specific exception you need handling of, such as conversion exception, TBD if any math out of range exception would apply here at the edge. You don't want to swallow a memory exception or the like by accident here.

cdwiegand replied:
Ok - I copied this from Round() and modified from there, so I will remove the try/catch as there is no math exception I would catch that should not bubble up to the calling program (i.e. you try to Ceil(Int64.MaxValue) or something really weird should probably fire an exception).

daviburg replied:
Yes for those specific case you want to catch the specific exception
Console.WriteLine(Math.Ceiling(Decimal.MaxValue)); however does not throw
https://dotnetfiddle.net/MsJ4Kq

daviburg replied:
Round(...)'s precedence... not great :-) should fix it if we have the time, it will be tricky to figure out all exceptions it used to catch to weed the good from the bad.

cdwiegand commented:
Updated code to reflect recommendations, tests pass with standard values from the original liquid website's own examples. Cleaned up code re: variables, also using trycatch to handle odd values (like the word "two" not being valid).

daviburg commented on src/DotLiquid/StandardFilters.cs:
```diff
@@ -650,6 +650,32 @@ public static object Round(object input, object places = null)
+        /// Rounds a decimal value up to the next integer, unless already the integer value, removing all decimal places 
+        /// </summary>
+        /// <param name="input"></param>
+        /// <returns>The rounded value; null if an exception have occured</returns>
```
Nit nit: occurred (two 'r')

daviburg replied:
oh and, an exception has occurred (not have)

daviburg replied:
Merging as-is as no response. Consider a supplemental PR for the tweaks.
